src/term_ctl.c

```diff
@@ -17,64 +17,203 @@
+            return (4 + FOREGROUND_INTENSITY);
+       case TERM_COLOR_BRIGHT_MAGENTA:
+            return (5 + FOREGROUND_INTENSITY);
+       case TERM_COLOR_BRIGHT_YELLOW:
+            return (6 + FOREGROUND_INTENSITY);
+       case TERM_COLOR_BRIGHT_WHITE:
+            return (7 + FOREGROUND_INTENSITY);
+    }
+    fprintf(stderr,"FATAL: No mapping for TERM_COLOR_x=%d (fore: %d)\n", color, fore);
+    return (0);
+}
+
+static void _term_set_color(BOOL fore, term_color_t color)
+{
+    WORD win_color;
+
+    if (!console_file)
+        return;
+
+    if (color == TERM_COLOR_RESET) {
+        active_fg = (console_info.wAttributes & 7);
+        active_bg = (console_info.wAttributes & ~7);
+    }
+    else if (fore) {
+        active_fg = _term_get_win_color(TRUE,color);
+    }
+    else if (color <= TERM_COLOR_WHITE) {
+        active_bg = 16 * _term_get_win_color(FALSE,color);
+    }
+    else
+        return;
+
+    win_color = active_bg + active_fg;
+
+    /* Hack: as WinCon does not have color-themes (as Linux have) and no 'TERM_COLOR_BRIGHT_x'
+     * codes are used, always use a high-intensity foreground color. This look best in
+     * CMD with the default black background color.
+     */
+    if (fore && color != TERM_COLOR_RESET)
+        win_color |= FOREGROUND_INTENSITY;
+
+    fflush(console_file);
+    SetConsoleTextAttribute(console_hnd, win_color);
+}
+
+/*
+ * Cleanup in case we got a SIGINT signal in the middle of a
+ * non-default colour output.
+ */
+static void _term_exit (void)
+{
+    if (console_hnd && console_hnd != INVALID_HANDLE_VALUE) {
+        fflush(console_file);
+        SetConsoleTextAttribute(console_hnd, console_info.wAttributes);
+    }
+}
+
+static BOOL _term_init(FILE *fp)
+{
+    int fd;
+
+    if (console_hnd)
+        return (!console_redirected);
+
+    fd = fileno(fp);
+    if (fd == STDOUT_FILENO) {
+        console_hnd = GetStdHandle(STD_OUTPUT_HANDLE);
+        console_file = fp;
+    }
+    else if (fd == STDERR_FILENO) {
+        console_hnd = GetStdHandle(STD_ERROR_HANDLE);
+        console_file = fp;
+    }
+    console_redirected = (console_hnd == INVALID_HANDLE_VALUE) ||
+                         (!GetConsoleScreenBufferInfo(console_hnd, &console_info)) ||
+                         (GetFileType(console_hnd) != FILE_TYPE_CHAR);
+
+    _term_set_color(FALSE, TERM_COLOR_RESET); /* Set 'active_fg' and 'active_bg' */
+    atexit(_term_exit);
+    return (!console_redirected);
+}
+#endif /* _WIN32 */
+
 int term_get_columns(int fd)
 {
-#ifndef _WIN32
+#ifdef _WIN32
+    if (fd == fileno(console_file)) { /* Should we 'assert()' for this? */
+       /*
+        * Call this again as the screen dimensions could have changes since
+        * we called '_term_init()'.
+        */
+       CONSOLE_SCREEN_BUFFER_INFO c_info;
+
+       GetConsoleScreenBufferInfo(console_hnd, &c_info);
+       return (c_info.srWindow.Right - c_info.srWindow.Left + 1);
+    }
+    return (INT_MAX);
```

Comment:
"80" sounds like a better fallback

Reply:
My thinking was that on redirection (`rtl_433.exe what-ever > foo`), there is no limitation. But what is the KV-output supposed to look like then?

Reply:
Exactly right on the redirection. But for column wrapping there needs to be a sensible limit. Something between 80 and 160 looks good. I'd say 80 is always a good limit.